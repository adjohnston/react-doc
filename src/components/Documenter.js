import React from 'react'
import Title from 'components/Title'
import Notes from 'components/Notes'
import Presets from 'components/Presets'

export default ({name, notes, presets, setPreset, children}) => (
  <section>
    <Title
      text={name}/>

    {children}

    <Notes
      notes={notes}/>

    <Presets
      setPreset={setPreset}
      presets={presets}/>
  </section>
)
