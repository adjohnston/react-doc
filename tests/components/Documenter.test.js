import React from 'react'
import renderer from 'react-test-renderer'
import Documenter from 'components/Documenter'

const Test = () => <div>Just a test</div>
const mockProps = {
  setExample: () => {},
  setProp: () => {},
  componentProps: {
    id: {
      type: 'string'
    },

    onClick: {
      type: 'func',
      required: true
    }
  },
  swatches: ['white', 'black', '#f35'],
  children: <Test />,
  name: 'Test',
  description: 'Just a simple test component',
  status: 'READY',
  notes: {
    'Developer notes': 'I am just a string',
    'Designer notes': <div>I use a component</div>
  },
  examples: {
    'Basic example': {
      id: 'basic-example-id'
    }
  }
}

test('it should render correctly', () => {
  const tree = renderer
    .create(<Documenter {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly without a description', () => {
  const {
    description,
    ...otherProps
  } = mockProps

  const tree = renderer
    .create(<Documenter {...otherProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly without a swatches', () => {
  const {
    swatches,
    ...otherProps
  } = mockProps

  const tree = renderer
    .create(<Documenter {...otherProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly without a notes', () => {
  const {
    notes,
    ...otherProps
  } = mockProps

  const tree = renderer
    .create(<Documenter {...otherProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly without a examples', () => {
  const {
    examples,
    ...otherProps
  } = mockProps

  const tree = renderer
    .create(<Documenter {...otherProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly without a status', () => {
  const {
    status,
    ...otherProps
  } = mockProps

  const tree = renderer
    .create(<Documenter {...otherProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
