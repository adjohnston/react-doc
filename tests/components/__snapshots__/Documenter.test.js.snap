// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <span
    className="stncl__status stncl__status--ready"
  >
    Ready
  </span>
  <section
    className="stncl__description"
  >
    Just a simple test component
  </section>
  <section
    className="stncl__body"
  >
    <div>
      <div>
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": null,
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "white",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "black",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#f35",
            }
          }
        />
      </div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2
        className="stncl__title"
      >
        Examples
      </h2>
      <div
        className="stncl__dropdown"
      >
        <select
          className="stncl__dropdown__select"
          onChange={[Function]}
          value="default"
        >
          <option>
            Basic example
          </option>
        </select>
        <div
          className="stncl__dropdown__arrow"
        />
      </div>
    </section>
    <section>
      <h2>
        Notes
      </h2>
      <div>
        <h3>
          Developer notes
        </h3>
        I am just a string
      </div>
      <div>
        <h3>
          Designer notes
        </h3>
        <div>
          I use a component
        </div>
      </div>
    </section>
  </section>
</section>
`;

exports[`it should render correctly without a description 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <span
    className="stncl__status stncl__status--ready"
  >
    Ready
  </span>
  <section
    className="stncl__body"
  >
    <div>
      <div>
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": null,
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "white",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "black",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#f35",
            }
          }
        />
      </div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2
        className="stncl__title"
      >
        Examples
      </h2>
      <div
        className="stncl__dropdown"
      >
        <select
          className="stncl__dropdown__select"
          onChange={[Function]}
          value="default"
        >
          <option>
            Basic example
          </option>
        </select>
        <div
          className="stncl__dropdown__arrow"
        />
      </div>
    </section>
    <section>
      <h2>
        Notes
      </h2>
      <div>
        <h3>
          Developer notes
        </h3>
        I am just a string
      </div>
      <div>
        <h3>
          Designer notes
        </h3>
        <div>
          I use a component
        </div>
      </div>
    </section>
  </section>
</section>
`;

exports[`it should render correctly without a examples 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <span
    className="stncl__status stncl__status--ready"
  >
    Ready
  </span>
  <section
    className="stncl__description"
  >
    Just a simple test component
  </section>
  <section
    className="stncl__body"
  >
    <div>
      <div>
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": null,
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "white",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "black",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#f35",
            }
          }
        />
      </div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2>
        Notes
      </h2>
      <div>
        <h3>
          Developer notes
        </h3>
        I am just a string
      </div>
      <div>
        <h3>
          Designer notes
        </h3>
        <div>
          I use a component
        </div>
      </div>
    </section>
  </section>
</section>
`;

exports[`it should render correctly without a notes 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <span
    className="stncl__status stncl__status--ready"
  >
    Ready
  </span>
  <section
    className="stncl__description"
  >
    Just a simple test component
  </section>
  <section
    className="stncl__body"
  >
    <div>
      <div>
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": null,
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "white",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "black",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#f35",
            }
          }
        />
      </div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2
        className="stncl__title"
      >
        Examples
      </h2>
      <div
        className="stncl__dropdown"
      >
        <select
          className="stncl__dropdown__select"
          onChange={[Function]}
          value="default"
        >
          <option>
            Basic example
          </option>
        </select>
        <div
          className="stncl__dropdown__arrow"
        />
      </div>
    </section>
  </section>
</section>
`;

exports[`it should render correctly without a status 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <section
    className="stncl__description"
  >
    Just a simple test component
  </section>
  <section
    className="stncl__body"
  >
    <div>
      <div>
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": null,
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "white",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "black",
            }
          }
        />
        <button
          className="stncl__swatch-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#f35",
            }
          }
        />
      </div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2
        className="stncl__title"
      >
        Examples
      </h2>
      <div
        className="stncl__dropdown"
      >
        <select
          className="stncl__dropdown__select"
          onChange={[Function]}
          value="default"
        >
          <option>
            Basic example
          </option>
        </select>
        <div
          className="stncl__dropdown__arrow"
        />
      </div>
    </section>
    <section>
      <h2>
        Notes
      </h2>
      <div>
        <h3>
          Developer notes
        </h3>
        I am just a string
      </div>
      <div>
        <h3>
          Designer notes
        </h3>
        <div>
          I use a component
        </div>
      </div>
    </section>
  </section>
</section>
`;

exports[`it should render correctly without a swatches 1`] = `
<section
  className="stncl__layout"
>
  <h1
    className="stncl__name"
  >
    Test
  </h1>
  <span
    className="stncl__status stncl__status--ready"
  >
    Ready
  </span>
  <section
    className="stncl__description"
  >
    Just a simple test component
  </section>
  <section
    className="stncl__body"
  >
    <div>
      <div
        className="stncl__preview"
        style={
          Object {
            "background": null,
          }
        }
      >
        <div>
          Just a test
        </div>
      </div>
    </div>
    <section>
      <h2
        className="stncl__title"
      >
        Examples
      </h2>
      <div
        className="stncl__dropdown"
      >
        <select
          className="stncl__dropdown__select"
          onChange={[Function]}
          value="default"
        >
          <option>
            Basic example
          </option>
        </select>
        <div
          className="stncl__dropdown__arrow"
        />
      </div>
    </section>
    <section>
      <h2>
        Notes
      </h2>
      <div>
        <h3>
          Developer notes
        </h3>
        I am just a string
      </div>
      <div>
        <h3>
          Designer notes
        </h3>
        <div>
          I use a component
        </div>
      </div>
    </section>
  </section>
</section>
`;
